test/shopify-cli/task/ensure_env_test.rb

```diff
@@ -14,18 +14,176 @@ def setup
         ShopifyCli::Tunnel.stubs(:start)
       end
 
-      def test_ask_strips_out_https_from_shop
-        CLI::UI.expects(:ask).times(3)
-          .returns('apikey', 'apisecret', 'https://test-shop.myshopify.com')
-        @task = EnsureEnv.call(@context)
-        assert_equal('test-shop.myshopify.com', Project.current.env.shop)
-      end
-
-      def test_writes_env_if_no_env
-        CLI::UI.expects(:ask).times(3)
-          .returns('apikey', 'apisecret', 'https://test-shop.myshopify.com')
-        @task = EnsureEnv.call(@context)
-        assert_equal('apikey', Project.current.env.api_key)
+      def test_create_new_app_if_none_available
+        ShopifyCli::PartnersAPI::Organizations.stubs(:fetch_with_app).returns([{
+          "id" => 421,
+          "businessName" => "one",
+          "stores" => [{
+            "shopDomain" => "store.myshopify.com",
+          }],
+          "apps" => [],
+        }])
+        CLI::UI::Prompt.expects(:ask).with(@context.message('core.tasks.ensure_env.app_name')).returns('new app')
+        CLI::UI::Prompt.expects(:ask).with(@context.message('core.tasks.ensure_env.app_type.select')).returns('public')
+        ShopifyCli::Tasks::CreateApiClient.expects(:call).with(
+          @context,
+          org_id: 421,
+          title: 'new app',
+          type: 'public',
+        ).returns({
+          "apiKey" => "ljdlkajfaljf",
+          "apiSecretKeys" => [{ "secret" => "kldjakljjkj" }],
+          "id" => "12345678",
+        })
+        Resources::EnvFile.any_instance.stubs(:write)
+        EnsureEnv.call(@context)
```

Comment:
For all the `.call`, we need to verify the return value contains the correct `org` hash.